java/source/io.rst

```diff
@@ -298,14 +303,56 @@ Read - From File
     ageVector.set(1, 20);
     ageVector.set(2, 30);
     vectorSchemaRoot.setRowCount(3);
+    Field name2 = new Field("name2", FieldType.nullable(new ArrowType.Utf8()), null);
+    Field age2 = new Field("age2", FieldType.nullable(new ArrowType.Int(32, true)), null);
+    Schema schemaPerson2 = new Schema(asList(name2, age2));
+    VectorSchemaRoot vectorSchemaRoot2 = VectorSchemaRoot.create(schemaPerson2, rootAllocator);
+    VarCharVector nameVector2 = (VarCharVector) vectorSchemaRoot2.getVector("name2");
+    nameVector2.allocateNew(3);
+    nameVector2.set(0, "Nidia".getBytes());
+    nameVector2.set(1, "Alexa".getBytes());
+    nameVector2.set(2, "Mara".getBytes());
+    IntVector ageVector2 = (IntVector) vectorSchemaRoot2.getVector("age2");
+    ageVector2.allocateNew(3);
+    ageVector2.set(0, 15);
+    ageVector2.set(1, 20);
+    ageVector2.set(2, 15);
+    vectorSchemaRoot2.setRowCount(3);
+    Field name3 = new Field("name3", FieldType.nullable(new ArrowType.Utf8()), null);
+    Field age3 = new Field("age3", FieldType.nullable(new ArrowType.Int(32, true)), null);
+    Schema schemaPerson3 = new Schema(asList(name3, age3));
+    VectorSchemaRoot vectorSchemaRoot3 = VectorSchemaRoot.create(schemaPerson3, rootAllocator);
+    VarCharVector nameVector3 = (VarCharVector) vectorSchemaRoot3.getVector("name3");
+    nameVector3.allocateNew(3);
+    nameVector3.set(0, "Raul".getBytes());
+    nameVector3.set(1, "Jhon".getBytes());
+    nameVector3.set(2, "Thomy".getBytes());
+    IntVector ageVector3 = (IntVector) vectorSchemaRoot3.getVector("age3");
+    ageVector3.allocateNew(3);
+    ageVector3.set(0, 34);
+    ageVector3.set(1, 29);
+    ageVector3.set(2, 33);
+    vectorSchemaRoot3.setRowCount(3);
     File file = new File("randon_access_to_file.arrow");
     try (FileOutputStream fileOutputStream = new FileOutputStream(file);
          ArrowFileWriter writer = new ArrowFileWriter(vectorSchemaRoot, null, /*WritableByteChannel out*/fileOutputStream.getChannel())
     ){
         // write
         writer.start();
-        for (int i=0; i<10; i++){
+        for (int i=0; i<3; i++){
             // Generate data or modify the root or use a VectorLoader to get fresh data from somewhere else
+            if (i==1){
+                VectorUnloader vectorUnloader2 = new VectorUnloader(vectorSchemaRoot2);
+                ArrowRecordBatch arrowRecordBatch2 = vectorUnloader2.getRecordBatch();
+                VectorLoader vectorLoader2 = new VectorLoader(vectorSchemaRoot);
```

Comment:
FWIW, why not clean things up and unload the batch when we first create the other roots?

Reply:
Though, a suggestion: why not just `ageVector.set(0, i * 10); ageVector.set(1, i * 10 + 1); ...` etc in the loop and demonstrate modifying the root?

Reply:
We could have a separate recipe to demonstrate VectorLoader/VectorUnloader.

Reply:
Added suggestion and also added a separate recipe to demonstrate VectorLoader/VectorUnloader.